Report errors returned by _qspi_configure_format

kyle-cypress opened the pull request:
### Description
The function returns a qspi_status_t but most usages in QSPIFBlockDevice assume that it always succeeds.

<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->


### Pull request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [x] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

<!--
    Optional
    Request additional reviewers with @username
-->

### Release Notes

<!--
    Optional
    In case of breaking changes, functionality changes or refactors, please add release notes here. 
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->

Commit:
Report errors returned by _qspi_configure_format

The function returns a qspi_status_t but most usages in QSPIFBlockDevice
assume that it always succeeds.

0xc0170 commented on components/storage/blockdevice/COMPONENT_QSPIF/QSPIFBlockDevice.h:
```diff
@@ -29,9 +29,10 @@ enum qspif_bd_error {
-    QSPIF_BD_ERROR_INVALID_ERASE_PARAMS  = -4005, /* Erase command not on sector aligned addresses or exceeds device size */
-    QSPIF_BD_ERROR_DEVICE_NOT_UNIQE      = -4006, /* Only one instance per csel is allowed */
-    QSPIF_BD_ERROR_DEVICE_MAX_EXCEED     = -4007 /* Max active QSPIF devices exceeded */
+    QSPIF_BD_ERROR_CONF_FORMAT_FAILED    = -4005, /* Configure format failed */
```
`QSPIF_BD_ERROR_CONF_FORMAT_FAILED    ` shall this be 4008? Or this follows `_FAILED` thus placed irght after the latest failure status and the rest is shifted.

kyle-cypress replied:
This is made moot by 5b3b147 which removes the new error code entirely.

0xc0170 reviewed (approved):
LGTM , just a question about error numbering

Edit: rethinking my review after the above comment, missed it prior reviewing

Commit:
Reuse existing error for _qspi_configure_format

Use QSPIF_BD_ERROR_DEVICE_ERROR instead of introducing a new error code.
Add tr_error calls whenever _qspi_configure_format fails to aid in debugging.

SeppoTakalo reviewed (approved):
LGTM

VeijoPesonen commented on components/storage/blockdevice/COMPONENT_QSPIF/QSPIFBlockDevice.cpp:
```diff
@@ -302,17 +306,23 @@ int QSPIFBlockDevice::read(void *buffer, bd_addr_t addr, bd_size_t size)
     _mutex.lock();
```
Remember to release this. ~Could you also check all the places where the code decides to jump to `goto exit_point`-label as the the same issue seems to be found also from other functions inside this file.~ Scratch that, other places seem to be fine.

kyle-cypress replied:
Updated to follow the `goto exit_point` pattern used everywhere else.

VeijoPesonen reviewed (changes requested):
Please see my previous comment.

Commit:
Avoid stale mutex in QSPIFBlockDevice::read

Update to follow the same `goto exit_point` pattern that is used
by the rest of the functions to avoid leaving the mutex locked
when errors are detected and require the function to abort.

VeijoPesonen reviewed (approved):
Looks good to me
